### ISSUE TITLE
invalid byte sequence in UTF-8

### ISSUE THREAD

**Issue by mariuspopaadrian** [rubocop/rubocop#10093]
## Expected behavior

to finish without (invalid byte sequence in UTF-8) error , tested also with latest version of rubocop

Describe here how you expected RuboCop to behave in this particular situation.

## Actual behavior

```
bundle exec rubocop --debug
For /home/ubuntu/work/interests: Default configuration from /home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from Loading cache from /home/ubuntu/.cache/rubocop_cache/7ef6358f605b5805babb7c5740600ded484e75e3/6d7a3b621ca1730e04accd938619e4bdab66cfb1/b63397466ceea1c7de9cd43823bcff9ed7716caf
....
invalid byte sequence in UTF-8
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parser-3.0.2.0/lib/parser/source/buffer.rb:167:in `source='
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cached_data.rb:49:in `deserialize_offenses'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cached_data.rb:14:in `from_json'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/result_cache.rb:110:in `load'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/runner.rb:145:in `file_offense_cache'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/runner.rb:129:in `file_offenses'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/runner.rb:67:in `block in warm_cache'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:515:in `call_with_index'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:485:in `process_incoming_jobs'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:465:in `block in worker'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:456:in `fork'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:456:in `worker'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:447:in `block in create_workers'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:446:in `each'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:446:in `each_with_index'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:446:in `create_workers'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:386:in `work_in_processes'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:289:in `map'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/parallel-1.21.0/lib/parallel.rb:238:in `each'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/runner.rb:67:in `warm_cache'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/runner.rb:46:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/command.rb:11:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli/environment.rb:18:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli.rb:71:in `run_command'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli.rb:78:in `execute_runners'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/lib/rubocop/cli.rb:47:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/exe/rubocop:12:in `block in <top (required)>'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/benchmark.rb:308:in `realtime'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/rubocop-1.19.0/exe/rubocop:12:in `<top (required)>'
/home/ubuntu/.rbenv/versions/3.0.2/bin/rubocop:23:in `load'
/home/ubuntu/.rbenv/versions/3.0.2/bin/rubocop:23:in `<top (required)>'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `load'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:63:in `kernel_load'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli/exec.rb:28:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:474:in `exec'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:30:in `dispatch'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/cli.rb:24:in `start'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.2.22/libexec/bundle:49:in `block in <top (required)>'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/3.0.0/bundler/friendly_errors.rb:128:in `with_friendly_errors'
/home/ubuntu/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/bundler-2.2.22/libexec/bundle:37:in `<top (required)>'
/home/ubuntu/.rbenv/versions/3.0.2/bin/bundle:23:in `load'
/home/ubuntu/.rbenv/versions/3.0.2/bin/bundle:23:in `<main>'
Finished in 0.27516150000155903 seconds
```

## Steps to reproduce the problem

if is needed i will try to find the offending file 
if i remove the cache , then it shows the cop offences 
rm -rf  /home/ubuntu/.cache/rubocop_cache
....
175 files inspected, 592 offenses detected, 407 offenses auto-correctable
invalid byte sequence in UTF-8
...
then the rest of the  error from above 

## RuboCop version

```
$ bundle exec rubocop -V
1.19.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-linux)
```

**Comment by dvandersluis** [comment]
It'd be good if you can run `rubocop -d -C false` to run in debug mode without the cache, and see if that gives more of an idea where the invalid byte sequence is coming from.

**Comment by mariuspopaadrian** [comment]
rubocop -d -C false | grep utf-8
test/models/device_token_test.rb:1:1: F: Lint/Syntax: Invalid byte sequence in utf-8.

this is the file with issues , i think is a compiled shader or something , not a ruby file 
`P_uvTransformedCoords_0_Stage0Rwebgl_FragColor`

**Comment by dvandersluis** [comment]
You should add that file to your `Exclude` configuration in .rubocop.yml or fix the invalid utf-8 in it.

```yaml
AllCops:
  Exclude:
    - test/modesl/device_token_test.rb
    - ...
```
